fdbclient/DatabaseConfiguration.cpp

```diff
@@ -540,3 +552,35 @@ void DatabaseConfiguration::makeConfigurationImmutable() {
 		rawConfiguration[i++] = KeyValueRef( rawConfiguration.arena(), KeyValueRef( r->first, r->second ) );
 	mutableConfiguration = Optional<std::map<std::string,std::string>>();
 }
+
+void DatabaseConfiguration::fromKeyValues(Standalone<VectorRef<KeyValueRef>> rawConfig) {
+	resetInternal();
+	this->rawConfiguration = rawConfig;
+	for (auto c = rawConfiguration.begin(); c != rawConfiguration.end(); ++c) {
+		setInternal(c->key, c->value);
+	}
+	setDefaultReplicationPolicy();
+}
+
+bool DatabaseConfiguration::isOverridden(const std::string& key) const {
+	for (auto iter = rawConfiguration.begin(); iter != rawConfiguration.end(); ++iter) {
+		auto confKey = iter->key.removePrefix(configKeysPrefix).toString();
+		if (key == confKey) {
+			return true;
+		}
+	}
+
+	if (!mutableConfiguration.present()) {
+		return false;
+	}
+
+	for (auto iter = mutableConfiguration.get().begin(); iter != mutableConfiguration.get().end(); ++iter) {
```

Comment:
If `mutableConfiguration` is present, then `rawConfiguration` is not valid. So this loop should be checked first, and previous loop on `rawConfiguration` should be conditional on `mutableConfiguration` is not present.